zkevm-circuits/src/evm_circuit/execution/codesize.rs

```diff
@@ -0,0 +1,134 @@
+use array_init::array_init;
+use bus_mapping::evm::OpcodeId;
+use eth_types::Field;
+use halo2_proofs::plonk::Error;
+
+use crate::{
+    evm_circuit::{
+        step::ExecutionState,
+        util::{
+            common_gadget::SameContextGadget,
+            constraint_builder::{ConstraintBuilder, StepStateTransition, Transition},
+            from_bytes, CachedRegion, Cell, RandomLinearCombination,
+        },
+        witness::{Block, Call, ExecStep, Transaction},
+    },
+    util::Expr,
+};
+
+use super::ExecutionGadget;
+
+#[derive(Clone, Debug)]
+pub(crate) struct CodesizeGadget<F> {
+    same_context: SameContextGadget<F>,
+    codesize_bytes: [Cell<F>; 8],
+    codesize: Cell<F>,
+}
+
+impl<F: Field> ExecutionGadget<F> for CodesizeGadget<F> {
+    const NAME: &'static str = "CODESIZE";
+
+    const EXECUTION_STATE: ExecutionState = ExecutionState::CODESIZE;
+
+    fn configure(cb: &mut ConstraintBuilder<F>) -> Self {
+        let opcode = cb.query_cell();
+
+        let codesize_bytes = array_init(|_| cb.query_cell());
```

Comment:
why not query_byte? 
We should constrain them really being byte

Reply:
Yes that's right, we do wish to constrain each one to be within a byte.